### PR TITLE
Zig Fmt recursive option

### PR DESCRIPTION
**Pull request by BraedonWooding**
As the title says, you can call it like `zig fmt src/ otherFile.zig otherStuff/inside/ -r`, if you forget the -r it'll give you an error `To fmt recursively pass the '-r' flag.`, this is to prevent people accidentally doing recursive and causing unexpected damage for whatever reason :).

Added this since I wanted to run it on a source directory of my Lazy Zig library and it was annoying to have to give 10+ source files rather than just saying `zig fmt -r src/`

NOTE: doesn't work on windows, since windows doesn't have directory code yet.  I'll have a look in another PR if I have time to see if I can implement it.

### CODE REVIEW

File: src-self-hosted/main.zig [hunk 1]
```diff
@@ -687,19 +687,61 @@ const usage_fmt =
...
 
+// Returns false if any errors occur
+fn fmtHandleFile(allocator: *Allocator, file_path: []const u8, color: errmsg.Color) !bool {
+    var file = try os.File.openRead(allocator, file_path);
+    defer file.close();
+
+    return blk: {
+        const source_code = io.readFileAlloc(allocator, file_path) catch |err| {
+            try stderr.print("unable to open '{}': {}\n", file_path, err);
+            break :blk false;
```
Comment (tiehuis) [inline src-self-hosted/main.zig]: You can just use a straight return for these. No need to pass return via the `:blk`.

**Comment by BraedonWooding** [comment]
@andrewrk the the failed travis check don't seem at all related to my PR maybe another issue from another PR?  And the other one is because windows directories isn't implemented yet in the std (`error: TODO support Dir.open for windows`), this is a separate issue to this implementation.

How would you advise we approach this?

**Comment by andrewrk** [comment]
This PR cannot be merged until #709 is done

**Comment by BraedonWooding** [comment]
When #1071 is pushed this can be merged in, I'll close and reopen it before then so the checks will re-run :).

**Comment by thejoshwolfe** [comment]
Recursive directory traversal is a dangerous road to go down. There are lots of features you can support in this domain that people will ask for. `grep` has 11 command line options for "File and Directory Selection" including features like exclude patterns, and following vs skipping symlinks.

Traversing directories is a harder problem than it seems, and I think we should leave it outside the scope of zig fmt. You can always use `find` to enumerate files and pass those to any other program using the `-exec` option.

**Comment by BraedonWooding** [comment]
@thejoshwolfe I fail to see how it is a 'bad' thing?  And yes there are a lot of features and I presume eventually most of them will be supported, almost every 'fmt'er out there supports this basic recursion, for example go [fmt](https://golang.org/cmd/gofmt/) supports this exact type of fmt I'm showing here (very simple recursive search).

And for the 'scope' of this, it really isn't that much larger; its performing an extremely simple recursive search on directories for the exact reason you gave, its a lot harder of a problem then it seems and it is easier to begin with a simple solution.  I don't think using find and -exec is a solution, that seems hackish and annoying to me personally :).

**Comment by thejoshwolfe** [comment]
This implementation selects files that end with `.zig`. That might be worth documenting.

**Comment by BraedonWooding** [comment]
I'll squash the commits once I've updated it (just away from computer right now)

**Comment by andrewrk** [comment]
I'll look at this tomorrow morning.

**Comment by andrewrk** [comment]
I haven't forgotten this - I'm part way through merging it, should have it merged tomorrow at some point.

**Pull request closed without merging**